e2e/settings.spec.ts

```diff
@@ -0,0 +1,97 @@
+import { expect } from '@playwright/test'
+import { test } from './base-test'
+
+test.use({
+  // TODO: import session value in auth.json from environment variables
+  // to match the SESSION_SECRET and MAGIC_LINK_SECRET in .env file
+  storageState: 'e2e/auth.json',
+})
+
+test('Validate phone number when updating data', async ({
+  page,
+  noscript,
+  queries: { getByRole },
+}) => {
+  // Go to http://localhost:3000/dashboard/settings
+  await page.goto('/dashboard/settings')
+  // Query phoneNumber
+  const phoneNumber = await getByRole('textbox', {
+    name: /nomor whatsapp/i,
+  })
+  // Fill phoneNumber
+  await phoneNumber.fill('6512345678')
+  // Press Tab
+  await phoneNumber.press('Tab')
+  // Expect visibility only when JavaScript is enabled
+  if (!noscript) {
+    await expect(
+      page
+        .locator(
+          'text=Nomor WhatsApp harus mengandung kode negara dan nomor telepon'
+        )
+        .first()
+    ).toBeVisible()
+  }
+  // Fill phoneNumber
+  await phoneNumber.fill('+6512345678')
+  // Click text=Save
+  await Promise.all([
+    page.waitForNavigation(/*{ url: 'http://localhost:3000/dashboard' }*/),
+    page.click('text=Simpan'),
+  ])
+  // Expect invisibility
+  await expect(
+    page
+      .locator(
+        'text=Nomor WhatsApp harus mengandung kode negara dan nomor telepon'
+      )
+      .first()
+  ).not.toBeVisible()
+})
+
+test('Validate name when updating data', async ({
+  page,
+  noscript,
+  queries: { getByRole },
+}) => {
+  await page.goto('/dashboard/settings')
+
+  const name = await getByRole('textbox', {
+    name: /nama lengkap/i,
+  })
+
+  await name.fill('')
+  await name.press('Tab')
+
+  if (!noscript) {
+    await expect(
+      page.locator('text=Nama Lengkap wajib diisi').first()
+    ).toBeVisible()
+  }
+
+  await name.fill('Lorem I')
+  await page.click('text=Simpan')
+
+  await expect(
+    page.locator('text=Nama Lengkap wajib diisi').first()
+  ).not.toBeVisible()
+})
+
+test('Update profile', async ({ page }) => {
+  await page.goto('/dashboard/settings')
+
+  await page.fill('[name="name"]', 'Lorem Ipsum')
+  await page.fill('[name="phoneNumber"]', '+6289123456')
+  await page.fill('[name="telegram"]', '@lorem_tl')
+  await page.fill('[name="instagram"]', '@lorem_ig')
```

Comment:
Please use `getByRole` query from `playwright-testing-library`.
You can use [Testing Playground](https://testing-playground.com)'s browser extensions to get the recommended query easily.

You can refer to [line 18 to 20](https://github.com/zainfathoni/kelas.rumahberbagi.com/pull/60/files?authenticity_token=l%2BWR4je6AIxYGVwXMLA5JSn8uzfsAljBhdUZCzLY9pekTkk1%2B7ib7SFNAMYBKhT0bc5MN3lkEAv4gJk7Hrenbw%3D%3D&file-filters%5B%5D=.ts&hide-deleted-files=true#diff-c1dd7f901a5034e85bf5308f50eefd5c35ec9c44c146eef8465355580a245e5bR18-R20) as an example.